api/src/endpoints/svgThumbnail.js

```diff
@@ -0,0 +1,44 @@
+import express from 'express';
+import fs from 'fs';
+import getSvgThumbnail from 'utils/2d-map';
+import models from 'data/integratedModels.json';
+
+const VALID_MODELS = models.map((m) => m.short_name);
+
+const routes = express.Router();
+const sharp = require('sharp');
+
+routes.get('/:svgName', async (req, res) => {
+  const { svgName } = req.params;
+  const { model, version, width } = req.query;
+
+  try {
+    if (!VALID_MODELS.includes(model)) {
+      throw new Error(`Invalid model provided: ${model}.`);
+    }
+
+    const min_w = 100;
+    const max_w = 600;
+    var w_thumb = 400;
+    const isnum = /^\d+$/.test(width);
+    if (!isnum) {
+      throw new Error(`Invalid width provided: ${width}. It should be an integer.`);
+    } else {
+      w_thumb =  parseInt(width) || 400;
```

Comment:
Is the intention of this line to have width set as 400 if no width query param is provided? The current behavior is that `Invalid width provided: undefined. It should be an integer.` if you don't provide a width query param

Reply:
It can be removed after I added `isnum` checking.